SPECS-SIGNED/mstflint-driver-signed/mstflint-signed.spec

```diff
@@ -0,0 +1,36 @@
+%define mstflint_unsigned_name mstflint-kernel
+
+Summary:        Mellanox firmware burning tool
+Name:           %{mstflint_unsigned_name}-signed
+Version:        4.21.0
+Release:        4%{?dist}
+License:        Dual BSD/GPL
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          System Environment/Kernel
+URL:            https://github.com/Mellanox/%{name}
+Source0:        https://github.com/Mellanox/%{name}/releases/download/v%{version}-1/%{name}-%{version}-1.tar.gz
+
+# set package name
+%{!?_name: %global _name mstflint}
+
+%global kver %(/bin/rpm -q --queryformat '%{RPMTAG_VERSION}-%{RPMTAG_RELEASE}' $(/bin/rpm -q --whatprovides kernel-devel))
+%global install_mod_dir extra/%{_name}
+
+%install
+export INSTALL_MOD_PATH=$RPM_BUILD_ROOT
+mkdir -p %{install_mod_dir}
+cd kernel
+install -d $INSTALL_MOD_PATH/lib/modules/%{kver}/%{install_mod_dir}
+cp $PWD/obj/mstflint_access.ko $INSTALL_MOD_PATH/lib/modules/%{kver}/%{install_mod_dir}
+cd -
+
```

Comment:
The usual way to handle kernel modules in Linux is through Dynamic Kernel Module Support (DKMS), and it is generally not recommended to install kernel modules manually.

PS: As for the Microsoft workflow and its relation to this project, I am not familiar with them and have no connections to this project. Please consider me as a random individual who has been asked to review a change in a project that I am not familiar with. However, I do find the installation of a kernel module object to be suspicious.

Reply:
Thanks for your comments. That's a fair point, however as Chaitanya mentioned it's not supported by Mariner ecosystem.